tests/test-distribution/test-ee10-distribution/src/test/java/org/eclipse/jetty/ee10/tests/distribution/SiweTests.java

```diff
@@ -0,0 +1,153 @@
+
+    @Test
+    public void testSiwe() throws Exception
+    {
+        Path jettyBase = newTestJettyBaseDirectory();
+        String jettyVersion = System.getProperty("jettyVersion");
+        JettyHomeTester distribution = JettyHomeTester.Builder.newInstance()
+            .jettyVersion(jettyVersion)
+            .jettyBase(jettyBase)
+            .build();
+
+        String[] args1 = {
+            "--create-startd",
+            "--approve-all-licenses",
+            "--add-to-start=http,ee10-webapp,ee10-deploy,ee10-annotations,siwe"
+        };
+
+        try (JettyHomeTester.Run run1 = distribution.start(args1))
+        {
+            assertTrue(run1.awaitFor(START_TIMEOUT, TimeUnit.SECONDS));
+            assertEquals(0, run1.getExitValue());
+
+            Path webApp = distribution.resolveArtifact("org.eclipse.jetty.ee10:jetty-ee10-test-siwe-webapp:war:" + jettyVersion);
+            distribution.installWar(webApp, "test");
+            Files.createDirectory(jettyBase.resolve("etc"));
+            Path realmProperties = Files.createFile(jettyBase.resolve("etc/realm.properties"));
+            try (FileWriter fw = new FileWriter(realmProperties.toFile()))
+            {
+                fw.write(_credentials.getAddress() + ":,admin\n");
+            }
+
+            int port = Tester.freePort();
+            String[] args2 = {
+                "jetty.http.port=" + port,
+                "jetty.ssl.port=" + port,
+                "jetty.server.dumpAfterStart=true",
+            };
+
+//            System.setProperty("distribution.debug.port", "5005");
+            try (JettyHomeTester.Run run2 = distribution.start(args2))
+            {
+                assertTrue(run2.awaitConsoleLogsFor("Started oejs.Server@", START_TIMEOUT, TimeUnit.SECONDS));
+                startHttpClient(false);
+                String uri = "http://localhost:" + port + "/test";
+
+                // Initially not authenticated.
+                ContentResponse response = client.GET(uri + "/");
+                assertThat(response.getStatus(), is(HttpStatus.OK_200));
+                String content = response.getContentAsString();
+                assertThat(content, containsString("not authenticated"));
+
+                // Request to /admin redirects to loginPage.
+                client.setFollowRedirects(false);
+                response = client.GET(uri + "/admin");
+                assertThat(response.getStatus(), is(HttpStatus.FOUND_302));
+                assertThat(response.getHeaders().get(HttpHeader.LOCATION),
+                    containsString(uri + "/login.html"));
+
+                // Fetch a nonce from the server.
+                response = client.GET(uri + "/auth/nonce");
+                String nonce = parseNonce(response.getContentAsString());
+                assertThat(nonce.length(), equalTo(8));
+
+                // Request to authenticate redirects to /admin page.
```

Comment:
Where's all this stuff about `MetaMask` in all this? Doesn't it have to be locally installed (as per the `login.html` page instructions?)

Reply:
The tests use `EthereumCredentials` which I wrote to generate an etherem address and sign messages programmatically without needing the browser libraries.

So no metamask needed.